readthedocs/oauth/management/commands/sync_vcs_data.py

```diff
@@ -0,0 +1,74 @@
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand
+
+from readthedocs.oauth.tasks import sync_remote_repositories
+
+
+class Command(BaseCommand):
+    help = "Sync OAuth RemoteRepository and RemoteOrganization"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--queue',
+            type=str,
+            default='resync-oauth',
+            help='Celery queue name.',
+        )
+        parser.add_argument(
+            '--users',
+            nargs='*',
+            type=str,
+            default=[],
+            help='Re-sync VCS provider data for specific users only.',
+        )
+        parser.add_argument(
+            '--skip-users',
+            nargs='*',
+            type=str,
+            default=[],
+            help='Skip re-sync VCS provider data for specific users.',
+        )
+        parser.add_argument(
+            '--max-users',
+            type=int,
+            default=100,
+            help='Maximum number of users that should be synced.',
+        )
+
+    def handle(self, *args, **options):
+        queue = options.get('queue')
+        sync_users = options.get('users')
+        skip_users = options.get('skip_users')
+        max_users = options.get('max_users')
+
+        # Filter users who have social accounts connected
+        # and has no remote repository relations
+        users = User.objects.filter(
+            socialaccount__isnull=False,
+            remote_repository_relations__isnull=True
+        ).distinct()
+
+        if sync_users:
+            users = users.filter(username__in=sync_users)
+
+        if skip_users:
+            users = users.exclude(username__in=skip_users)
+
+        users_to_sync = users.values_list('id', flat=True)[:max_users]
+
+        self.stdout.write(
+            self.style.SUCCESS(
+                'Found %s user(s) with the given parameters' % users.count()
+            )
+        )
```

Comment:
Can we also show the total number of users that need to be resync before taking the `max_users` slice?

Reply:
`users.count()` shows the total number of users that needs to be re-synced according to the filters. The next count is the count after the slice. Did you mean something else?

Reply:
Yeah, I wanted to know the total of users that needs to be synced without applying `sync_users` or `skip_users`. So, maybe those filters need to be applied to `users_to_sync` instead of `users`. Don't you think?

In that case we will have:

- Found TOTAL user(s) to sync in total
- Re-syncing VCS Providers for FILTERED_USERS user(s) with given parameters

This gives us a clear number about "where we are" when running this command. Every time we run it we know exactly how many users still need re-syncing.